src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Url/Base64UrlDecoder.cs

```diff
@@ -0,0 +1,794 @@
+        /// </remarks>
+        public static OperationStatus DecodeFromUtf8(ReadOnlySpan<byte> source, Span<byte> destination, out int bytesConsumed, out int bytesWritten, bool isFinalBlock = true) =>
+            DecodeFrom<Base64UrlDecoderByte, byte>(source, destination, out bytesConsumed, out bytesWritten, isFinalBlock, ignoreWhiteSpace: true);
+
+        /// <summary>
+        /// Decode the span of UTF-8 encoded text in Base64Url (in-place) into binary data.
+        /// The decoded binary output is smaller than the text data contained in the input (the operation deflates the data).
+        /// </summary>
+        /// <param name="buffer">The input span which contains the base 64 text data that needs to be decoded.</param>
+        /// <returns>The number of bytes written into the <paramref name="buffer"/>. This can be used to slice the output for subsequent calls, if necessary.</returns>
+        /// <exception cref="FormatException"><paramref name="buffer"/> contains an invalid Base64Url character,
+        /// more than two padding characters, or a non-white space-character among the padding characters.</exception>
+        /// <remarks>
+        /// As padding is optional the input length not required to be a multiple of 4.
+        /// If the input length is not a multiple of 4 the remainders decoded accordingly:
+        /// Remainder of 3 bytes - decoded into 2 bytes data, decoding succeeds.
+        /// Remainder of 2 bytes - decoded into 1 byte data. decoding succeeds.
+        /// Remainder of 1 byte - is invalid input, causes FormatException.
+        /// </remarks>
+        public static int DecodeFromUtf8InPlace(Span<byte> buffer)
+        {
+            OperationStatus status = DecodeFromUtf8InPlace<Base64UrlDecoderByte>(buffer, out int bytesWritten, ignoreWhiteSpace: true);
+
+            // Base64.DecodeFromUtf8InPlace returns OperationStatus, therefore doesn't throw.
+            // For Base64Url, this is not an OperationStatus API and thus throws.
+            if (status == OperationStatus.InvalidData)
+            {
+                throw new FormatException(SR.Format_BadBase64Char);
+            }
+
+            return bytesWritten;
+        }
+
+        /// <summary>
+        /// Decode the span of UTF-8 encoded text represented as Base64Url into binary data.
+        /// </summary>
+        /// <param name="source">The input span which contains UTF-8 encoded text in Base64Url that needs to be decoded.</param>
+        /// <param name="destination">The output span which contains the result of the operation, i.e. the decoded binary data.</param>
+        /// <returns>The number of bytes written into the <paramref name="destination"/>. This can be used to slice the output for subsequent calls, if necessary.</returns>
+        /// <exception cref="ArgumentException">Thrown when the encoded output cannot fit in the <paramref name="destination"/> provided.</exception>
```

Comment:
https://github.com/dotnet/dotnet-api-docs/wiki/Exceptions says all descriptions should be written as if preceded by "[Exception] is thrown if ", except that it starts with a capital letter.

```suggestion
        /// <exception cref="ArgumentException">The encoded output cannot fit in the destination provided.</exception>
```

To match some other span writing docs I know of:

```suggestion
        /// <exception cref="ArgumentException">The buffer in <paramref name="destination"/> is too small to hold the encoded output.</exception>
```